src/main/java/com/mixpanel/android/mpmetrics/MPDbAdapter.java

```diff
@@ -102,6 +102,13 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
             db.execSQL(PEOPLE_TIME_INDEX);
         }
 
+        public boolean belowMemThreshold() {
```

Comment:
In that case, perhaps we should put all of the "database full" logic in MPDbAdapter? Perhaps by simply ignoring the write (and logging an error) in addJSON if the memory is too full? That would mean that MPDbAdapter was responsible for not only the "How big is too big" part of the policy, but also the "what to do if you're too big" part of the policy.